README.md

```diff
@@ -44,10 +45,162 @@ thrift_files: Mix.Utils.extract_files(["thrift"], [:thrift])
+`get_user_by_id_with_options/3` | Allows you to pass `gen_tcp` and `GenServer` options to your client. This is useful for setting the `GenServer` timeout if you expect your RPC to take longer than the default of 5 seconds. Like `get_user_by_id/2`, this function returns `{:ok, response}` or `{:error, reason}` tuples.
+`get_user_by_id_with_options!/3` | Allows you to pass `gen_tcp` and `GenServer` options and raises an exception if an error occurs. 
+
+#### Note: in the above example, the function `deleteUser` will be converted to `delete_user` to comply with Elixr's naming conventions.
+
+To use the client, simply call `start_link`, supplying the host and port. 
+
+```elixir
+alias Thrift.Test.UserService.Clients.Binary.Framed, as: Client
+{:ok, client} = UserService.Clients.Binary.Framed.start_link("localhost", 2345, [])
+
+iex(1)> {:ok, user} = Client.get_user_by_id(client, 22451)
+{:ok, %Thrift.Test.User{id: 22451, username: "stinky", first_name: "Stinky", last_name: "Stinkman"}}
+``` 
+
+The client supports the following options:
+
+Option name      |  Type | Description
+-----------------|-------|-------------
+`:tcp_opts` | keyword | A keyword list of tcp options
+`:gen_server_opts` | keyword | A keyword list of options for the gen server 
+
+##### TCP Opts 
+
+Name             | Type | Description 
+-----------------|------|---------------
+`:timeout`       | positive integer | The default timeout for reading from, writing to and connecting to sockets.
+`send_timeout`   | positive integer | The amount of time in milliseconds to wait before sending data fails.
+`backoff_calculator` | (int) -> int | A single argument function that takes the number of retries and returns the amount of time to wait in milliseconds before reconnecting. The default implementation waits 100, 100, 200, 300, 500, 800 and then 1000 ms. All retries after that will wait 1000ms.
 
-## Thrift IDL Parsing
 
-This package also contains experimental support for parsing [Thrift IDL][idl]
+##### GenServer Opts 
+Name             | Type | Description 
+-----------------|------|---------------
+`timeout`        | A positive integer | The amount of time in milliseconds the Client's GenServer waits for a reply. After this, the GenServer will exit with `{:error, :timeout}`.
+
+
+### Example of using options
+
+```elixir
+alias Thrift.Test.UserService.Clients.Binary.Framed, as: Client
+{:ok, client} = UserService.Clients.Binary.Framed.start_link("localhost", 2345, 
+                tcp_opts: [backoff_calculator: fn(retry_count) -> retry_count * 1000], gen_server_opts: [timeout: 10_000])
+                
+```
+In the above example, the client will use a (very) conservative backoff calculator that waits an additional second each time the client retries. In other words, the first retry will wait 1 second, the second will wait 2 seconds, and the third will wait 3. It sets the GenServer timeout to be ten seconds, which means the remote side can take its time to reply. 
```

Comment:
"It sets the GenServer timeout" => "These options set the GenServer timeout" (ambiguous pronoun)